### PR TITLE
feat: Support recursive listing into Files API #337

### PR DESCRIPTION
**Pull request by astsiapanay**

### CODE REVIEW

File: src/main/java/com/epam/aidial/core/controller/FileMetadataController.java [hunk 1]
```diff
@@ -27,9 +27,12 @@ private String getContentType() {
     @Override
     protected Future<?> handle(ResourceDescription resource) {
         BlobStorage storage = proxy.getStorage();
+        boolean recursive = Boolean.parseBoolean(context.getRequest().getParam("recursive", "false"));
+        String token = context.getRequest().getParam("token");
+        int limit = Integer.parseInt(context.getRequest().getParam("limit", "100"));
         return proxy.getVertx().executeBlocking(() -> {
             try {
-                MetadataBase metadata = storage.listMetadata(resource);
+                MetadataBase metadata = storage.listMetadata(resource, token, limit, recursive);
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/controller/FileMetadataController.java:38]: MetadataBase does not contain, next token parameter

File: src/main/java/com/epam/aidial/core/controller/FileMetadataController.java [hunk 2]
```diff
@@ -27,9 +27,12 @@ private String getContentType() {
     @Override
     protected Future<?> handle(ResourceDescription resource) {
         BlobStorage storage = proxy.getStorage();
+        boolean recursive = Boolean.parseBoolean(context.getRequest().getParam("recursive", "false"));
+        String token = context.getRequest().getParam("token");
+        int limit = Integer.parseInt(context.getRequest().getParam("limit", "100"));
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/controller/FileMetadataController.java:32]: please validate limit query parameter the same way like in ResourceController

Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/controller/FileMetadataController.java:32]: fixed

File: src/main/java/com/epam/aidial/core/controller/FileMetadataController.java (hunk 1 above)
Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/controller/FileMetadataController.java:38]: fixed

File: src/main/java/com/epam/aidial/core/controller/FileMetadataController.java [hunk 3]
```diff
@@ -27,9 +27,16 @@ private String getContentType() {
...
+        if (limit < 0 || limit > 1000) {
+            context.respond(HttpStatus.BAD_REQUEST, "Limit is out of allowed range: [0, 1000]");
+            return Future.succeededFuture();
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/controller/FileMetadataController.java]: Nit, `return context.respond(....`

**Review by Maxim-Gadalov — DISMISSED** [review]

File: src/main/java/com/epam/aidial/core/controller/FileMetadataController.java (hunk 3 above)
Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/controller/FileMetadataController.java]: fixed

**Review by Maxim-Gadalov — APPROVED** [review]

**Comment by astsiapanay** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-355.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**